test/Microsoft.Azure.Devices.Edge.Test.Common/certs/CertificateAuthority.cs

```diff
@@ -2,16 +2,20 @@
 namespace Microsoft.Azure.Devices.Edge.Test.Common.Certs
 {
     using System;
+    using System.IO;
     using System.Threading;
     using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Edge.Util;
+    using Serilog;
     using RootCaKeys = System.ValueTuple<string, string, string>;
 
     public class CertificateAuthority
     {
         readonly Option<string> scriptPath;
 
-        public EdgeCertificates Certificates { get; }
+        public EdgeCertificates EdgeCertificate { get; }
+
+        public Certificates Certificate { get; }
```

Comment:
- - - - - - - - - - - - -
`CreateDpsX509CaAsync()` is not equivalent to 
```
CertificateAuthority.CreateAsync()
ca.GenerateDeviceCertificatesAsync()
```
`CreateDpsX509CaAsync()` does the following: 
1. Generate & Install Root cert (if not existed)
2. Generate & Install Device cert given the Root cert
while the proposed changes would
1. Generate & Install Root cert (always)
2. Generate & Install Edge (server) cert
3. Generate & Install Device cert
Hence, the proposed change would introduce another set of certs into the test. 
 - - - - - - - - - - - - -
As of `CertificateAuthority` class, my understanding of the term is an entity that manages all the certificates being use. That's why I decided it to be a composition class composes of `EdgeCertificate` and `Certificate`
 - - - - - - - - - - - - -
`GenerateLeafCertificatesAsync(2)` is basically `GenerateDeviceCertificatesAsync(3)` with two input arguments instead of 3. I didn't change the `GenerateLeafCertificatesAsync(2)` name to limit the scope of change.

Reply:
Oh I see what you mean. I'll address your comment: 
```
Previously we had "edge certificates" and "leaf certificates". Edge certs were really CA certs, and leaf certs were really "identity" (aka end-entity) certs. With your new test, this distinction isn't enough because now we have edge CA certs and edge identity certs (as well as leaf identity certs). Maybe the names for the two cert classes should be CaCertificate and IdentityCertificate.
CertificateAuthority would return these certificate objects from GenerateCaCertificate() and GenerateIdentityCertificate() respectively.
```

Reply:
> my understanding of the term is an entity that manages all the certificates being use ...

A certificate authority does not _manage_ certs; it _issues_ certs.

> CreateDpsX509CaAsync() is not equivalent to ...

I guess the part I left out of my original comment was that `CertificateAuthority` would need an additional (non-static) method that generates a CA certificate based on the root cert.

So the setup fixture for the `TransparentGateway` test would call:
```
CertificateAuthority.CreateAsync()
ca.GenerateCaCertificateAsync()
```

...while the `DpsX509Test` test would call:
```
CertificateAuthority.CreateAsync()
ca.GenerateDeviceCertificatesAsync()
```